packages/flame_3d/lib/src/resources/material/spatial_material.dart

```diff
@@ -61,29 +59,40 @@ class SpatialMaterial extends Material {
 
   @override
   void bind(GraphicsDevice device) {
+    _bindVertexInfo(device);
+    _bindMaterialProperties();
+    _bindLights(device);
+    _bindCamera(device);
+  }
+
+  void _bindVertexInfo(GraphicsDevice device) {
     vertexShader
       ..setMatrix4('VertexInfo.model', device.model)
       ..setMatrix4('VertexInfo.view', device.view)
       ..setMatrix4('VertexInfo.projection', device.projection);
+  }
 
-    final invertedView = Matrix4.inverted(device.view);
-
-    const radius = 15;
-    final angle = DateTime.now().millisecondsSinceEpoch / 4000;
-    final x = cos(angle) * radius;
-    final z = sin(angle) * radius;
-
+  void _bindMaterialProperties() {
     fragmentShader
-      // Material
       ..setTexture('albedoTexture', albedoTexture)
       ..setVector3('Material.albedoColor', _albedoCache)
       ..setFloat('Material.metallic', metallic)
       ..setFloat('Material.metallicSpecular', metallicSpecular)
-      ..setFloat('Material.roughness', roughness)
-      // Light
-      ..setVector3('Light.position', Vector3(x, 10, z))
-      // Camera
-      ..setVector3('Camera.position', invertedView.transform3(Vector3.zero()));
+      ..setFloat('Material.roughness', roughness);
+  }
+
+  void _bindCamera(GraphicsDevice device) {
+    final invertedView = Matrix4.inverted(device.view);
+    final cameraPosition = invertedView.transform3(Vector3.zero());
+    fragmentShader.setVector3('Camera.position', cameraPosition);
+  }
+
+  void _bindLights(GraphicsDevice device) {
+    final light = device.lights.firstOrNull;
+    if (light == null) {
+      return;
+    }
+    light.apply(fragmentShader);
```

Comment:
So are we applying or binding, the method name indicates something else. 

I vote for naming it `_applyLights` and calling it in the `_bindMaterials` instead. Because in the end it gets applied to the shader of the material.

Reply:
I can rename to _applyLights, or I can rename light.apply to light.bind - no preference whatsoever
however I don't know if this should be inside the _bindMaterials, feels like light is a separate concept that the shader can use for several things... not super strong opinion though

Reply:
The spatial material is light aware and will always apply lights on it's material.

Reply:
I know, I was just thinking that other parts of the shader might want to know about light information, seemed more like a global concept than something that belongs in the material. but I can move it inside if you prefer, doesn't make any difference

Reply:
in fact it is not even inside the `Material.` namespace, it has it's own `Light.` namespace... that is how it is currently organized